test/validation/api/atomic/atomic.c

```diff
@@ -706,6 +710,114 @@ static void test_atomic_non_relaxed_128_acq_rel(void)
 	}
 }
 
+static void test_atomic_bit_set_32(void)
+{
+	const uint32_t bits = 1U << (odp_thread_id() - 1);
```

Comment:
I noticed now that the way thread_id() is used as a bit position is a bit problematic. The bit shifting by thread_id() becomes UB (and likely breaks the test too) as soon as MAX_WORKERS is made bigger than 32. Another problem is that the bit operations do not test the high order half of 64 bit values, so the test would not notice if my 64-bit bit ops are just 32-bit ops.

I think both of these problems could be fixed by deriving the bit position from thread ID somehow like this: bits = 1UL << ((odp_thread_id() * 33) % 64) (and similarly for the 32-bit values).

Reply:
Is "* 33" somehow specifically important here? If it is, I should probably add a comment about it in the code.

Reply:
It changes the order of the bits tested to avoidf the problem that with a limited number of threads only the lowest orders bits are tested. With that, the high order half of the 64 bit atomic gets tested already with two workers. Actually, 41 would be even better than 33 as it would distribute the bits better with e.g. 4 threads.

Reply:
It would of course be better to have that in a function, e.g. thread_id_to_bit_pattern_u64().

Reply:
OK, neat. Refactored in v3.